pkg/maps/policymap/policymap.go

```diff
@@ -26,19 +26,19 @@ import (
 )
 
 const (
-	// CallMapName is the name of the map to do tail calls into policy
-	// enforcement programs
-	CallMapName = "cilium_policy"
+	// PolicyCallMapName is the name of the map to do tail calls into policy
+	// enforcement programs.
+	PolicyCallMapName = "cilium_call_policy"
```

Comment:
I assume this won't brake up and downgrade because we always create the map when we initialized the map?

Reply:
My understanding is that we (tc) repopulate the map when we reload the datapath, so this won't be an issue for up/downgrades. I expect the old map to remain behind though; I don't know if we usually clean this sort of stale BPF objects (?)

Reply:
My inclination is to agree with @pchaigno here, but I think it'd be worthwhile to run some tests on this. For instance set up a couple of chatty peers, perform upgrade, and check that they continue chatting. We could probably use one of the chaos monkeys for this, eg:
https://github.com/cilium/chaos-monkeys/tree/master/monkeys/clique

(Worth double-checking the behaviour by deploying the monkey, restarting an existing version of Cilium first to understand how it performs, then try upgrade and observe)

> I don't know if we usually clean this sort of stale BPF objects (?)

Yes, we clean the old objects. I think there's proxymap references still in the code to help clean up from upgrade from older versions, maybe v1.5 or so. Of course, if we downgrade again then we'll leave the map with the new name on the filesystem... though I'm less concerned about that since we assume users will eventually get onto a stable version of the newer release.